### ISSUE TITLE
Eclipse with TestNG and maven-surefire-plugin

### ISSUE THREAD

**Issue by avaysberg** [cbeust/testng-eclipse#186]
Hi,
ich have a Problem with run TestNG(Plugin: 6.9.9.201510270734) in Eclipse v. 4.5.1:
I have in pom.xml:
```xml
 <artifactId>maven-surefire-plugin</artifactId>
                <configuration>
                    <properties>
                        <property>
                            <name>usedefaultlisteners</name>
                            <!-- disabling default listeners is optional -->
                            <value>false</value>
                        </property>
                        <property>
                            <name>listener</name>
                            <value>org.testng.reporters.JUnitReportReporter,org.testng.reporters.JUnitReportReporter,
                                org.testng.reporters.jq.Main,org.testng.reporters.FailedReporter,org.testng.reporters.EmailableReporter</value>
                        </property>
                    </properties>
                    <argLine>
                        -javaagent:"${settings.localRepository}"/org/jmockit/jmockit/${jmockit-version}/jmockit-${jmockit-version}.jar
                    </argLine>
                    <systemPropertyVariables>
                        <jacoco-agent.destfile>${basedir}/target/jacoco.exec</jacoco-agent.destfile>
                    </systemPropertyVariables>
                    <suiteXmlFiles>
                        <suiteXmlFile>${basedir}/target/test-classes/testng.xml</suiteXmlFile>
                    </suiteXmlFiles>
                </configuration>
```
and when I run the test with testng-eclipse plugin I receive the error: Reference to undefined variable settings.localRepository. It seems that the testng-plugin doesn't see this setting. 
On Eclipse 4.3.1 and 4.5.1 and Plugin TestNG 6.9.5.201508210528 it works without any problem.

**Comment by missedone** [comment]
@avaysberg 
to solve this issue, you can install the simple plugin here:
https://github.com/missedone/m2e-ext/releases

it will resolve `${settings.localRepository}` with the value you configured in m2e eclipse plugin, normally it's `$HOME/.m2/repository`

**Comment by avaysberg** [comment]
1. Ok, from v. 6.9.8 you support args. But doesn't support resolving from maven properties. When do you want that support. Or it will be support on separate plugin? 

2. the release exist for https://github.com/missedone/m2e-ext/releases, but doesn't describe how it install.  In this case i let use old 6.9.5

**Comment by avaysberg** [comment]
Can you args support for maven-surefire and other with checkbox disable or enable management. It's allow no additional maven extension installation.

**Comment by missedone** [comment]
for now i made it as a separate plugin, however it is meant to be submitted to testng plugin but since it will make testng plugin wired with m2e plugin which i don't feel conformable with it.
i will investigate if i can make m2e as an optional dependency later.

regards to the option allowing you disable/enable this behavior, you can found it on run configuration page. if you require for a workspace level preference, unfortunate it's not available at the moment, i can add one if you demand.
thanks

**Comment by missedone** [comment]
@avaysberg 
to instal the [m2e-ext](https://github.com/missedone/m2e-ext/releases) plugin, you can follow this:
http://stackoverflow.com/a/16074606/4867232

**Comment by avaysberg** [comment]
I think on workspace level preference, were it my solution.
Thank you very much for smart support!

**Comment by missedone** [comment]
the m2e integration is done in PR #191, people can disable/enable this feature at workspace level preference, later on I will update the docs for TestNG Eclipse Plugin.

**Comment by missedone** [comment]
this is now well implemented in #193

**Comment by missedone** [comment]
Edit: please see the usage docs here: http://testng.org/doc/eclipse.html#eclipse-m2e-integration
